### PR TITLE
Consistent rounding rules for float display with fixed precision

### PR DESCRIPTION
**Pull request by ayushmishra2005**
#70336

### CODE REVIEW

**Review by dtolnay — CHANGES_REQUESTED** [review]
I haven't gotten a chance to review the implementation yet but it looks like there is some test code commented out in flt2dec/mod.rs that probably shouldn't be. Is that on purpose? What's the deal with those tests?

File: src/test/ui/ifmt.rs [hunk 1]
```diff
@@ -144,10 +144,10 @@ pub fn main() {
...
     t!(format!("{:e}", 1.2345e6f64), "1.2345e6");
     t!(format!("{:E}", 1.2345e6f64), "1.2345E6");
-    t!(format!("{:.3e}", 1.2345e6f64), "1.234e6");
-    t!(format!("{:10.3e}", 1.2345e6f64),   "   1.234e6");
-    t!(format!("{:+10.3e}", 1.2345e6f64),  "  +1.234e6");
-    t!(format!("{:+10.3e}", -1.2345e6f64), "  -1.234e6");
+    t!(format!("{:.3e}", 1.2345e6f64), "1.235e6");
+    t!(format!("{:10.3e}", 1.2345e6f64),   "   1.235e6");
+    t!(format!("{:+10.3e}", 1.2345e6f64),  "  +1.235e6");
+    t!(format!("{:+10.3e}", -1.2345e6f64), "  -1.235e6");
```
Comment (dtolnay) [inline src/test/ui/ifmt.rs:150]: It looks to me like these were correct before. Round-to-even on 1234.5 should go to 1234, not 1235.

**Comment by crlf0710** [comment]
@ayushmishra2005 Ping from triage, would you mind addressing the review comments above? thanks

**Comment by ayushmishra2005** [comment]
@crlf0710   I am closing this pull request for now because I don't have the bandwidth to address them. I will raise new PR after addressing the above comments.

**Pull request closed without merging**